internal/pkg/flags/flags.go

```diff
@@ -302,32 +309,39 @@ type FlagVar struct {
 func (f FlagVar) Validate() error {
 	if f.EnableKcpWatcher {
 		if f.WatcherImageTag == "" {
-			return errMissingWatcherImageTag
+			return ErrMissingWatcherImageTag
+		}
+		if f.WatcherImageRegistry == "" {
+			return ErrMissingWatcherImageRegistry
 		}
 		dirInfo, err := os.Stat(f.WatcherResourcesPath)
 		if err != nil || !dirInfo.IsDir() {
-			return errWatcherDirNotExist
+			return ErrWatcherDirNotExist
 		}
 	}
 
 	if f.LeaderElectionRenewDeadline >= f.LeaderElectionLeaseDuration {
-		return fmt.Errorf("%w (%.1f[s])", errLeaderElectionTimeoutConfig, f.LeaderElectionLeaseDuration.Seconds())
+		return fmt.Errorf("%w (%.1f[s])", ErrLeaderElectionTimeoutConfig, f.LeaderElectionLeaseDuration.Seconds())
 	}
 
 	if !map[int]bool{
 		2048: false, // 2048 is a valid value for cert-manager, but explicitly prohibited as not compliant to security requirements
 		4096: true,
 		8192: false, // see https://github.com/kyma-project/lifecycle-manager/issues/1793
 	}[f.SelfSignedCertKeySize] {
-		return errInvalidSelfSignedCertKeyLength
+		return ErrInvalidSelfSignedCertKeyLength
 	}
 
 	if f.ManifestRequeueJitterProbability < 0 || f.ManifestRequeueJitterProbability > 0.05 {
-		return errInvalidManifestRequeueJitterPercentage
+		return ErrInvalidManifestRequeueJitterPercentage
 	}
-	if f.ManifestRequeueJitterProbability < 0 || f.ManifestRequeueJitterProbability > 1 {
-		return errInvalidManifestRequeueJitterProbability
+	if f.ManifestRequeueJitterPercentage < 0 || f.ManifestRequeueJitterPercentage > 1 {
+		return ErrInvalidManifestRequeueJitterProbability
```

Comment:
to reviewer: this seems to be a bug from the previous implementation. There were two checks for **probability** and no one for **percentage**